src/main.c

```diff
@@ -872,6 +873,10 @@ static void keyer_init() {
 
     if (cwkeyer == HAMLIB_KEYER) {
 	showmsg("CW-Keyer is Hamlib");
+	if (trx_control_disabled) {
+	    showmsg("Radio control disabled - no keying!");
+	    return;
+	}
 	if (!trx_control) {
 	    showmsg("Radio control is not activated!!");
```

Comment:
Can we now reach this message? I.e. trx_control_disabled==false and trx_control==false

Reply:
Hmm, that state could only happen if you have RIGCONTROL disabled in logcfg.dat. Than having HAMLIB_KEYER activated seems to be an configuration error which is catched in the following if-statement. But I think that problem should be catched in parse_config() before.

Reply:
Indeed. Then the second `if` ("Radio control is not activated") has to be kept, even if only this quite specific use case triggers it. I'd not bother adding any other check for it.

On the other hand it may be worth promoting the message of the first `if` to a warning by adding a sleep. Now this message just flashed for a split second unless using verbose mode. Unless of course if we rely on the previous question (if the rig was off) or assume the user knows what they they are doing (if started with -r option).

Reply:
Good points. Give me a day or two to think about.

Reply:
Just added a 2 second delay (see commit below). should be enough to see a problem and restart with -v if needed.